RestAPI/Controllers/EmployeeController.cs

```diff
@@ -0,0 +1,160 @@
+﻿using System.Runtime.CompilerServices;
+using DTOs;
+using GrpcClient;
+using Microsoft.AspNetCore.Mvc;
+using RepositoryContracts;
+using Exception = System.Exception;
+using Microsoft.EntityFrameworkCore;
+using Employee = Entities.Employee;
+using EmployeeDTO = DTOs.EmployeeDTO;
+using NewEmployeeDTO = DTOs.NewEmployeeDTO;
+using Shift = Entities.Shift;
+using ShiftDTO = DTOs.Shift.ShiftDTO;
+using UpdateEmployeeDTO = GrpcClient.UpdateEmployeeDTO;
+namespace RestAPI.Controllers;
+
+[ApiController]
+[Route("[controller]")] 
+
+public class EmployeeController : ControllerBase
+{
+    private readonly IEmployeeRepository employeeRepo;
+    
+
+    public EmployeeController(IEmployeeRepository employeeRepository)
+    {
+        employeeRepo = employeeRepository;
+    }
+
+//this is the explanation --->             method is commented out as it was replaced by AuthController.cs/Register                
+    // [HttpPost]
+    // public async Task<ActionResult<SimpleEmployeeDTO>> AddEmployee([FromBody] NewEmployeeDTO request)
+    // {
+    //     try
+    //     {
+    //         Employee employee = await employeeRepo.AddAsync(EmployeeGrpcRepository.EntityNewEmployeeDtoToEntityEmployee(request));
+    //         var simpleDto = new SimpleEmployeeDTO
+    //         {
+    //             FirstName = employee.FirstName,
+    //             LastName = employee.LastName,
+    //             WorkingNumber = employee.WorkingNumber, 
+    //             Id = employee.Id
+    //         };
+    //         
+    //         return Ok(simpleDto);
+    //
+    //     }
+    //     catch(ArgumentException e)
+    //     {
+    //         return BadRequest(e.Message);
+    //     }
+    // }
+
+
+    [HttpPut("/Employee/{id:int}")] //problem: shifts are erased when user is updated
+    public async Task<ActionResult<PublicEmployeeDTO>> UpdateEmployee([FromRoute] int id, [FromBody] DTOs.UpdateEmployeeDTO request)
+    {
+        try
+        {
+            //List<Shift> shifts = new List<Shift>();
```

Comment:
was it what was causing issues with disappearing shifts on update?

Reply:
i have no idea we can delete this